src/envoy/prototype/api_manager_filter.cc

```diff
@@ -275,31 +283,35 @@ class Instance : public Http::StreamFilter,
 
   virtual FilterHeadersStatus encodeHeaders(HeaderMap& headers,
                                             bool end_stream) override {
-    log().debug("Called ApiManager::Instance : {}", __func__);
+    Log().debug("Called ApiManager::Instance : {}", __func__);
     return FilterHeadersStatus::Continue;
   }
   virtual FilterDataStatus encodeData(Buffer::Instance& data,
                                       bool end_stream) override {
-    log().debug("Called ApiManager::Instance : {}", __func__);
+    Log().debug("Called ApiManager::Instance : {}", __func__);
     return FilterDataStatus::Continue;
   }
   virtual FilterTrailersStatus encodeTrailers(HeaderMap& trailers) override {
-    log().debug("Called ApiManager::Instance : {}", __func__);
+    Log().debug("Called ApiManager::Instance : {}", __func__);
     return FilterTrailersStatus::Continue;
   }
   virtual void setEncoderFilterCallbacks(
       StreamEncoderFilterCallbacks& callbacks) override {
-    log().debug("Called ApiManager::Instance : {}", __func__);
+    Log().debug("Called ApiManager::Instance : {}", __func__);
     encoder_callbacks_ = &callbacks;
   }
 
-  // note: cannot extend ~ActiveStream for access log, placing it here
-  ~Instance() {
-    log().debug("Called ApiManager::Instance : {}", __func__);
-    std::unique_ptr<google::api_manager::Response> response(new Response());
-    request_handler_->Report(std::move(response),
-                             [this]() { log().debug("Report returns"); });
+  virtual void log(const HeaderMap* request_headers,
+                   const HeaderMap* response_headers,
+                   const AccessLog::RequestInfo& request_info) override {
+    Log().debug("Called ApiManager::Instance : {}", __func__);
+    std::unique_ptr<google::api_manager::Response> response(
+        new Response(request_info));
+    request_handler_->Report(std::move(response), []() {});
   }
+  // There are two log() functions. Use this Log() to redirect to base class
+  // log().
+  spdlog::logger& Log() { return Logger::Loggable<Logger::Id::http>::log(); }
```

Comment:
hmm... you don't need this since `log()` signature is different, or if you want `Log()`, then probably we don't even need to inherit `Logger::Loggable`, just get the logger object from `Logger::Registry`